spring-cloud-contract-tools/spring-cloud-contract-gradle-plugin/src/main/groovy/org/springframework/cloud/contract/verifier/plugin/PublishStubsToScmTask.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2013-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.contract.verifier.plugin;
+
+import javax.inject.Inject;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.provider.MapProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputDirectory;
+import org.gradle.api.tasks.Nested;
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.PathSensitive;
+import org.gradle.api.tasks.PathSensitivity;
+import org.gradle.api.tasks.TaskAction;
+import org.springframework.cloud.contract.stubrunner.ContractProjectUpdater;
+import org.springframework.cloud.contract.stubrunner.ScmStubDownloaderBuilder;
+import org.springframework.cloud.contract.stubrunner.StubRunnerOptions;
+import org.springframework.cloud.contract.stubrunner.StubRunnerOptionsBuilder;
+import org.springframework.cloud.contract.stubrunner.spring.StubRunnerProperties;
+import org.springframework.util.StringUtils;
+
+/**
+ * For SCM based repositories will copy the generated stubs
+ * to the cloned repo with contracts and stubs. Will also
+ * commit the changes and push them to origin.
+ *
+ * NOTE: starting with 2.3.0.RELEASE the <code>customize{}</code> closure previously used for
+ * {@link PublishStubsToScmTask} customisation is no longer available. The settings should be applied directly
+ * within the <code>publishStubsToScm</code> closure as in the example above.
+ *
+ * @author Marcin Grzejszczak
+ * @author Anatoliy Balakirev
+ * @author Shannon Pamperl
+ * @since 2.0.0
+ */
+class PublishStubsToScmTask extends DefaultTask {
+
+	static final String TASK_NAME = "publishStubsToScm";
+
+	private final Repository contractRepository;
+
+	private final Property<StubRunnerProperties.StubsMode> contractsMode;
+
+	// FIXME: deleting the stubs after tests breaks incremental builds
+	private final Property<Boolean> deleteStubsAfterTest;
```

Comment:
So I reviewed this one more thoroughly and found it's usage to be safe from the original Gradle up-to-date checks. The premise is that it's purging the cloned Git repository from a temporary folder.

I think it's name could definitely be better or we could simply just configure it to true and not have the parameter.

Thoughts, @marcingrzejszczak?